library/core/src/ptr/const_ptr.rs

```diff
@@ -1333,6 +1331,14 @@ impl<T: ?Sized> *const T {
     /// beyond the allocation that the pointer points into. It is up to the caller to ensure that
     /// the returned offset is correct in all terms other than alignment.
     ///
+    /// When this is called during compile-time evaluation (which is unstable), the implementation
+    /// may return `usize::MAX` in cases where that can never happen at runtime. This is because the
```

Comment:
I'm happy to see that there's a question about this on the tracking issue (#90962), even if that resolves as-is.